feature_selection/variance_threshold_select.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pandas as pd
+
+def variance_threshold_select(data, threshold = 0):
+    """
+    Select features above a certain threshold of variance
+
+    Parameters
+    ----------
+    data : numpy ndarray, pandas DataFrame, list
+      A numpy array, a pandas DataFrame or list to select features from
+    threshold : float, optional
+      A variance threshold to filter features for
+
+    Returns
+    -------
+    numpy ndarray
+      A 1d array of indexes of the features that pass the threshold or
+      are not numerical
+
+    Examples
+    --------
+    >>> from feature_selection import variance_threshold_select
+    >>> X = [[1, 6, 0, 5], [1, 2, 4, 5], [1, 7, 8, 5]]
+    >>> variance_threshold_select(X)
+    array([1, 2])
+    """
+
+    is_data_list = isinstance(data, list)
+    is_data_df = isinstance(data, pd.DataFrame)
+    is_data_np_array = isinstance(data, np.ndarray)
+
+    if not (is_data_list or is_data_df or is_data_np_array):
+        raise Exception('Data is of an invalid type.')
```

Comment:
Let's use `TypeError` for consistency with scikit-learn

```suggestion
        raise TypeError('Data is of an invalid type.')
```